compiler/lexer/lexer.go

```diff
@@ -264,12 +264,29 @@ func (l *Lexer) resetNosymbol() {
 
 }
 
+// prevent double underscores and underscore at the end such as 1__2, 3_
+func (l *Lexer) checkNumberError() bool {
+
+	return l.ch == '_' && (l.readPosition >= len(l.input) || !isDigit(l.input[l.readPosition]))
+
+}
+
 func (l *Lexer) readNumber() []rune {
-	position := l.position
-	for isDigit(l.ch) {
+	var input []rune
+
+	for isDigit(l.ch) || l.ch == '_' {
+
+		if isDigit(l.ch) {
```

Comment:
IMO I think the lexer should preserve the underscores in the token. In terms of responsibility, I don't think a lexer should be responsible for any code transformation. Removing them at this stage seems sort of magical. 

IMO the token should be simplified at a later stage, maybe during parsing or eval?